dev/run-tests.py

```diff
@@ -0,0 +1,413 @@
+        sys.exit(e.returncode)
+
+
+def set_sbt_maven_profile_args():
+    """Properly sets the SBT environment variable arguments with additional
+    checks to determine if this is running on an Amplab Jenkins machine"""
+
+    # base environment values for SBT_MAVEN_PROFILE_ARGS_ENV which will be appended on
+    sbt_maven_profile_args_base = ["-Pkinesis-asl"]
+
+    sbt_maven_profile_arg_dict = {
+        "hadoop1.0" : ["-Dhadoop.version=1.0.4"],
+        "hadoop2.0" : ["-Dhadoop.version=2.0.0-mr1-cdh4.1.1"],
+        "hadoop2.2" : ["-Pyarn", "-Phadoop-2.2", "-Dhadoop.version=2.2.0"],
+        "hadoop2.3" : ["-Pyarn", "-Phadoop-2.3", "-Dhadoop.version=2.3.0"],
+    }
+
+    # set the SBT maven build profile argument environment variable and ensure
+    # we build against the right version of Hadoop
+    if os.environ.get("AMPLAB_JENKINS_BUILD_PROFILE"):
+        os.environ[SBT_MAVEN_PROFILE_ARGS_ENV] = \
+            " ".join(sbt_maven_profile_arg_dict.get(ajbp, []) 
+                     + sbt_maven_profile_args_base)
+    else:
+        os.environ[SBT_MAVEN_PROFILE_ARGS_ENV] = \
+            " ".join(sbt_maven_profile_arg_dict.get("hadoop2.3", [])
+                     + sbt_maven_profile_args_base)
+
+
+def is_exe(path):
+    """Check if a given path is an executable file
+    - from: http://stackoverflow.com/a/377028"""
+
+    return os.path.isfile(path) and os.access(path, os.X_OK)
+
+
+def which(program):
+    """Find and return the given program by its absolute path or 'None'
+    - from: http://stackoverflow.com/a/377028"""
+
+    fpath, fname = os.path.split(program)
+
+    if fpath:
+        if is_exe(program):
+            return program
+    else:
+        for path in os.environ.get("PATH").split(os.pathsep):
+            path = path.strip('"')
+            exe_file = os.path.join(path, program)
+            if is_exe(exe_file):
+                return exe_file
+    return None
+
+
+def determine_java_executable():
+    """Will return the *best* path possible for a 'java' executable or `None`"""
+
+    java_home = os.environ.get("JAVA_HOME")
+
+    # check if there is an executable at $JAVA_HOME/bin/java
+    java_exe = which(os.path.join(java_home, "bin/java"))
+    # if the java_exe wasn't set, check for a `java` version on the $PATH
+    return java_exe if java_exe else which("java")
+
+
+def determine_java_version(java_exe):
+    """Given a valid java executable will return its version in tuple format as:
+    [<major-version>, <minor-version>, <patch-version>, <update-version>]"""
+
+    raw_output = subprocess.check_output([java_exe, "-version"], stderr=subprocess.STDOUT)
+    raw_version_str = raw_output.split('\n')[0] # eg 'java version "1.8.0_25"'
+    version_str = raw_version_str.split()[-1].strip('"') # eg '1.8.0_25'
+    version, update = version_str.split('_') # eg ['1.8.0', '25']
+
+    # map over the values and convert them to integers
+    return map(lambda x: int(x), version.split('.') + [update])
```

Comment:
This is optional but definitely a solid step up in readability:

```python
from collections import namedtuple

JavaVersion = namedtuple('JavaVersion', ['major', 'minor', 'patch', 'update'])
return JavaVersion(major=, minor=, ...)
```

Then below, you can say this, which is much clearer:

```python
    if java_version.minor < 8:
        print "[warn] Java 8 tests will not run because JDK version is < 1.8."
```